### DISCUSSION TITLE
How to Encrypt .ovpn config file in apk

### DISCUSSION THREAD

**Discussion by FxamarAboali** [schwabe/ics-openvpn#1715] [Q&A] [answered]
How to Encrypt .ovpn config file data in apk  to secure it if my phone stolen.

and I want to know where config file store by default.

thank you

**Comment by schwabe** [comment, marked as answer]
The files of this app are already encrypted by default and not stored in plaintext. They are encrypt with a key using the keychain of the phone. The encrypted config files are in the default app date location.

**Reply by FxamarAboali** [reply]
ok, can I change file place or encrypt file data again.

**Reply by schwabe** [reply]
This is not a feature that is implemeted. I also not sure I understand the use case for that.

**Comment by FxamarAboali** [comment]
> The files of this app are already encrypted by default and not stored in plaintext. They are encrypt with a key using the keychain of the phone. The encrypted config files are in the default app date location.

what king of cypher encrypt the config file

**Reply by schwabe** [reply]
I am using this library https://developer.android.com/jetpack/androidx/releases/security and that uses AES in GCM mode to encrypt the files. See ProfileEncryption.kt for more details.
